src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -1155,7 +1155,7 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 			}
 		} else {
 			connection.connectHandler(false, info.errorMessage, info.errorNumber, info.messages);
-			this._telemetryService.createErrorEvent(TelemetryKeys.TelemetryView.Shell, TelemetryKeys.TelemetryError.DatabaseConnectionError, info.errorNumber.toString())
+			this._telemetryService.createErrorEvent(TelemetryKeys.TelemetryView.Shell, TelemetryKeys.TelemetryError.DatabaseConnectionError, info.errorNumber?.toString())
```

Comment:
Do you know why errorNumber is undefined in this case? This is fine to fix what is clearly an issue, but it may just be hiding some other problem so worth trying to dig further in to.

Reply:
(I would expect that if we're sending an error event we're always sending *something* in the errorNumber field. Even if it's just an empty string)

Reply:
At the moment no... but it's definitely in my mind. I have also noticed this when OE randomly fails, it returns session=false with no error information..